keras_nlp/src/models/backbone.py

```diff
@@ -107,11 +103,20 @@ def token_embedding(self, value):
     def get_config(self):
         # Don't chain to super here. `get_config()` for functional models is
         # a nested layer config and cannot be passed to Backbone constructors.
-        return {
+        config = {
             "name": self.name,
             "trainable": self.trainable,
         }
 
+        # Add quantization support by utilizing `DTypePolicyMap`
```

Comment:
This is great! This should buy us support for all models right? If possible we should consider extending our common backbones tests for this...

https://github.com/keras-team/keras-nlp/blob/e4f09b24c699857edae27c8054aab44078e9cbd5/keras_nlp/src/tests/test_case.py#L359-L367

https://github.com/keras-team/keras-nlp/blob/e4f09b24c699857edae27c8054aab44078e9cbd5/keras_nlp/src/models/gemma/gemma_backbone_test.py#L39-L45

Doing so would test quantization for the whole library. Seems like it should be doable, call quantize, asset output. WDYT?

If we run into failures for certain models, we could add an option to `run_backbone_test`, called `run_quantization_check=True`, and set the option to false if the model fails, with a TODO to investigate.

Reply:
Yeah, it is doable.
I have added `run_quantization_test` to `run_backbone_test`. Only Bloom and OPT failed the test.
However, there is a significant speed regression after adding this test. The CI time increased from ~19mins to ~27mins. Is this acceptable?

Reply:
I think having the coverage is important. Let's pull this in, and see if we can improve the runtime efficiency as a follow up.

Saving is slow. So maybe we can just do something like

Something like:
- Basic quantization tests do not hit saving. Just test `get_config()`, `from_config()` maybe assigning weights over.
- Separate quantization testing in our saving test harness. That is marked with `large`, and is only run on larger/faster hardware.

Reply:
Will try this in another PR.